tests/unit_tests/repobee/plugin_tests/test_github.py

```diff
@@ -578,3 +589,16 @@ def test_sets_assignees_defaults_to_notset(self, happy_github, api):
         impl_mock.create_issue.assert_called_once_with(
             "Title", body="Body", assignees=github.GithubObject.NotSet
         )
+
+
+class TestForOrganization:
+    """Tests for the for_organization function."""
+
+    def test_correctly_sets_provided_org(self, happy_github, api):
+        """Test that the provided organization is respected."""
+        new_org_name = "some-other-org"
+        mock_org = create_mock_organization(happy_github, new_org_name, [])
+        assert api.org != new_org_name
+        new_api = api.for_organization(new_org_name)
+
+        assert new_api.org is mock_org
```

Comment:
Maybe the order for these lines feels a bit wonky for me who is less experienced 

```python
def test_correctly_sets_provided_org(self, happy_github, api):
    """Test that the provided organization is respected."""
    new_org_name = "some-other-org"
    assert api.org != new_org_name
    
    mock_org = create_mock_organization(happy_github, new_org_name, [])
    new_api = api.for_organization(new_org_name)
    assert new_api.org is mock_org
```

But maybe part of the point is making sure that 

```python
mock_org = create_mock_organization(happy_github, new_org_name, [])
```

doesn't modify `api.org`?

Reply:
> But maybe part of the point is making sure that doesn't modify api.org?

No, just to verify that that it's actually a new name. But now looking at that, `api.org != new_org_name` doesn't even make sense as `api.org` returns an organization, while `new_org_name` is a string. /facepalm

Reply:
Does the revised test make more sense to you?